sst/include/SimengMemInterface.hh

```diff
@@ -0,0 +1,210 @@
+ public:
+  SimengMemInterface(StandardMem* mem, uint64_t cl, uint64_t max_addr,
+                     SST::Output* out);
+  /** Send Simeng's processImage to SST memory backend during `init` lifecycle
+   * phase of SST */
+  void sendProcessImageToSST(const span<char> image);
+
+  /**
+   * Construct an AggregatedReadRequest and use it to generate
+   * SST::StandardMem::Read request(s). These request(s) are then sent to SST.
+   */
+  void requestRead(const MemoryAccessTarget& target, uint64_t requestId = 0);
+
+  /**
+   * Construct an AggregatedWriteRequest and use it to generate
+   * SST::StandardMem::Write request(s). These request(s) are then sent to SST.
+   */
+  void requestWrite(const MemoryAccessTarget& target,
+                    const RegisterValue& data);
+
+  /** Retrieve all completed read requests. */
+  const span<MemoryReadResult> getCompletedReads() const;
+
+  /** Clear the completed reads. */
+  void clearCompletedReads();
+
+  /** Returns true if there are any oustanding memory requests */
+  bool hasPendingRequests() const;
+
+  /**
+   * Tick the memory interface to process SimEng related tasks. Since all memory
+   * operations are handled by SST this method is only used increment
+   * `tickCounter`.
+   */
+  void tick();
+
+  /**
+   * An instance of `SimengMemHanders` is registered to an instance of
+   * SST::StandardMem and is used to handle Read and Write response. The same
+   * instance of SST::StandardMem is passed to `SimengMemHandlers` to access
+   * private variables needed to handle responses correctly. Defining
+   * `SimengMemHandlers` as a friend class gives it access to all private
+   * variables defined in `SimengMemInterface`.
+   */
+  class SimengMemHandlers : public StandardMem::RequestHandler {
+    friend class SimengMemInterface;
+
+   public:
+    SimengMemHandlers(SimengMemInterface& interface, SST::Output* out)
+        : StandardMem::RequestHandler(out), mem_interface(interface) {}
+    ~SimengMemHandlers() {}
+    /**
+     * Overloaded instance of handle method to handle read request responses
+     * overriden to aggregate responses and send them back to SimEng.
+     */
+    void handle(StandardMem::ReadResp* resp) override;
+    /**
+     * Overloaded instance of handle method to handle write request responses
+     * overriden to delete the incoming responses as SimEng does not have any
+     * use for it.
+     */
+    void handle(StandardMem::WriteResp* resp) override;
+    SimengMemInterface& mem_interface;
+  };
+
+  /**
+   * This struct represents a memory request from SimEng, It is used as base
+   * struct for AggregateWriteRequest and AggregateReadRequest.
+   */
+  struct SimengMemoryRequest {
+    const MemoryAccessTarget target;
+
+    SimengMemoryRequest() : target(MemoryAccessTarget()){};
+    SimengMemoryRequest(const MemoryAccessTarget& target) : target(target){};
+  };
+
+  /**
+   * Structs AggregatedWriteRequest and AggregatedReadRequest are used to store
```

Comment:
Split into two separate DOXYGEN comments for these structs. Having it like this will only link this comment to the top struct and the bottom struct won't have anything.